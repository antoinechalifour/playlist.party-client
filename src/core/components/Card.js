import styled from 'styled-components'

const Card = styled.div`
  background: #fff;
  box-shadow: 0 1px 6px rgba(0, 0, 0, .25);
  border-radius: 4px;
  box-sizing: border-box;
`

export const CardContent = styled.div`
  padding: 16px;
  color: ${({ theme }) => theme.colors.typography};
`

export const CardActions = styled.div`
  text-align: right;
  padding: 24px 0;

  > * + * {
    margin-left: 8px;
  }
`

export default Card
