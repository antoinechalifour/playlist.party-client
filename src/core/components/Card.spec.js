import React from 'react'
import renderer from 'react-test-renderer'
import Card, { CardContent, CardActions } from './Card'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'

const theme = {
  colors: {
    dark: '#151515',
    danger: '#a63d40',
    primary: '#1db954',
    confirm: '#90a959',
    accent: '#6494aa',
    typography: '#373d3f',
    typographyLight: '#bcbcbc'
  }
}

describe('Card', () => {
  it('Should render a Card', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Card>
            <CardContent>
              <p>This is my test card content</p>
            </CardContent>
            <CardActions>
              <button>Click me!</button>
            </CardActions>
          </Card>
        </ThemeProvider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
