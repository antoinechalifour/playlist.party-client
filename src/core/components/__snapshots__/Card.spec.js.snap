// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should render a Card 1`] = `
.c0 {
  background: #fff;
  box-shadow: 0 1px 6px rgba(0,0,0,.25);
  border-radius: 4px;
  box-sizing: border-box;
}

.c1 {
  padding: 16px;
  color: #373d3f;
}

.c2 {
  text-align: right;
  padding: 24px 0;
}

.c2 > * + * {
  margin-left: 8px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <p>
      This is my test card content
    </p>
  </div>
  <div
    className="c2"
  >
    <button>
      Click me!
    </button>
  </div>
</div>
`;
